Source/Core/VideoCommon/TextureDecoder.h

```diff
@@ -14,94 +14,101 @@ enum
 {
-  // These are the texture formats that can be read by the texture mapper.
-  GX_TF_I4 = 0x0,
-  GX_TF_I8 = 0x1,
-  GX_TF_IA4 = 0x2,
-  GX_TF_IA8 = 0x3,
-  GX_TF_RGB565 = 0x4,
-  GX_TF_RGB5A3 = 0x5,
-  GX_TF_RGBA8 = 0x6,
-  GX_TF_C4 = 0x8,
-  GX_TF_C8 = 0x9,
-  GX_TF_C14X2 = 0xA,
-  GX_TF_CMPR = 0xE,
-
-  _GX_TF_ZTF = 0x10,  // flag for Z texture formats (used internally by dolphin)
-
-  // Depth texture formats (which directly map to the equivalent colour format above.)
-  GX_TF_Z8 = 0x1 | _GX_TF_ZTF,
-  GX_TF_Z16 = 0x3 | _GX_TF_ZTF,
-  GX_TF_Z24X8 = 0x6 | _GX_TF_ZTF,
-
-  _GX_TF_CTF = 0x20,  // flag for copy-texture-format only (used internally by dolphin)
-
-  // These are extra formats that can be used when copying from efb,
-  // they use one of texel formats from above, but pack diffrent data into them.
-  GX_CTF_R4 = 0x0 | _GX_TF_CTF,
-  GX_CTF_RA4 = 0x2 | _GX_TF_CTF,
-  GX_CTF_RA8 = 0x3 | _GX_TF_CTF,
-  GX_CTF_YUVA8 = 0x6 | _GX_TF_CTF,  // YUV 4:4:4 - Dolphin doesn't implement this format as no
-                                    // commercial games use it
-  GX_CTF_A8 = 0x7 | _GX_TF_CTF,
-  GX_CTF_R8 = 0x8 | _GX_TF_CTF,
-  GX_CTF_G8 = 0x9 | _GX_TF_CTF,
-  GX_CTF_B8 = 0xA | _GX_TF_CTF,
-  GX_CTF_RG8 = 0xB | _GX_TF_CTF,
-  GX_CTF_GB8 = 0xC | _GX_TF_CTF,
-
-  // extra depth texture formats that can be used for efb copies.
-  GX_CTF_Z4 = 0x0 | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z8H = 0x8 | _GX_TF_ZTF | _GX_TF_CTF,  // This produces an identical result to to GX_TF_Z8
-  GX_CTF_Z8M = 0x9 | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z8L = 0xA | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z16R = 0xB | _GX_TF_ZTF | _GX_TF_CTF,  // Reversed version of GX_TF_Z16
-  GX_CTF_Z16L = 0xC | _GX_TF_ZTF | _GX_TF_CTF,
+  // These values represent texture format in GX registers.
+  I4 = 0x0,
+  I8 = 0x1,
+  IA4 = 0x2,
+  IA8 = 0x3,
+  RGB565 = 0x4,
+  RGB5A3 = 0x5,
+  RGBA8 = 0x6,
+  C4 = 0x8,
+  C8 = 0x9,
+  C14X2 = 0xA,
+  CMPR = 0xE,
 };
 
-enum TlutFormat
+static inline bool IsColorIndexed(TextureFormat format)
 {
-  GX_TL_IA8 = 0x0,
-  GX_TL_RGB565 = 0x1,
-  GX_TL_RGB5A3 = 0x2,
+  return format == TextureFormat::C4 || format == TextureFormat::C8 ||
+         format == TextureFormat::C14X2;
+}
+
+// The EFB Copy pipeline looks like:
+//
+//   1. Read EFB -> 2. Select color/depth -> 3. Downscale (optional)
+//   -> 4. YUV conversion (optional) -> 5. Encode Tiles -> 6. Write RAM
+//
+// The "Encode Tiles" stage receives RGBA8 texels from previous stages and encodes them to various
+// formats. EFBCopyFormat is the tile encoder mode. Note that the tile encoder does not care about
+// color vs. depth or intensity formats - it only sees RGBA8 texels.
+enum class EFBCopyFormat
+{
+  // These values represent EFB copy format in GX registers.
+  // Most (but not all) of these values correspond to values of TextureFormat.
+  R4 = 0x0,  // R4, I4, Z4
+
+  // FIXME: Does 0x1 (Z8) have identical results to 0x8 (Z8H)?
+  //        Is either or both of 0x1 and 0x8 used in games?
```

Comment:
If we think about how the hardware is structured, would it be reasonable to expect that the 0x1 depth format interpreter the depth value as RGB colors, converts to YUV, and returns Y?